### DISCUSSION TITLE
2-phase or 3-phase flow?

### DISCUSSION THREAD

**Discussion by Bearunner** [idaholab/moose#25900] [Q&A Modules: Porous Flow] [answered]
Dear all,

I have a question:

Can the porous_flow module simulate 3-phase flow including a solid phase?

For example, if we get a precipitated solid phase by using the geochemistry module, the data from the geochemistry module will be transferred to the porous_flow module for flow calculation. Can the porous_flow module simulate precipitated solid-phase flow or not?

Best regards,

B

**Comment by Bearunner** [comment]
dear @GiudGiud and @cpgr , could you please help me with this? thanks

**Reply by cpgr** [reply]
You can have precipitation leading to porosity reduction and permeability impairment. Is that what you mean?

**Reply by Bearunner** [reply]
Sorry, it is not. What I want to know is whether this precipitation will flow with fluid or not.

**Reply by Bearunner** [reply]
Another question: can the porous flow module simulate 3-phase flow including gas-liquid-solid?

**Reply by cpgr** [reply, marked as answer]
So you want the solid phase to flow with the liquid phase I assume, with the same relative permeability? The only way I think it will do that currently is if you can express the solid phase as a liquid component - so converting the solid volume to a mass fraction of liquid volume. 

All of the physics (kernels etc) are extremely general, so can handle an arbitrary number of fluid phases, with an arbitrary number of fluid components per phase. The difficulty is getting all of the fluid properties, ref perms etc correct for your model.

**Reply by Bearunner** [reply]
Sorry, the flow of the solid phase has nothing to do with the flow of the liquid phase. Can the porous flow module simulate 3-phase flow including gas-liquid-solid seperately?

**Reply by cpgr** [reply]
I think you would have to add your own material to model this
